### DISCUSSION TITLE
How to place 30 items each page (arrange in 2 columns) ?

### DISCUSSION THREAD

**Discussion by EZSWTool01** [QuestPDF/QuestPDF#21] [Q&A] [answered]
Within the content of a page, I am trying to lay out 30 items in 2 columns: Item 1 to Item 15 in the left column, Item 16 to Item 30 in the right column. When there is over 30 items, a new page break will occur, with item 31 printed on the next page (in the left column)
Attached is a sample PDF file with the desired layout. 
I am not sure how to manage the flow of items across pages
[Content listing 30 items per page.pdf](https://github.com/QuestPDF/library/files/7080129/Content.listing.30.items.per.page.pdf)

**Comment by MarcinZiabek** [comment]
Hi, I have one potential solution in mind. I will share it tomorrow. 

Can you please provide your real use case so I can better understand your needs and requirements? Every detail will help a lot.

I am looking for the real-world example where such layout is needed, assuming that the document is dynamically generated.

**Comment by EZSWTool01** [comment]
It is for a packing slip. For instance, after packing items in a box, you
need to print a packing slip listing the serial number of items packed. The
serial numbers can be printed in barcode format. This will make it easy for
the shipper to scan and verify whether the items are included in a box.
Since there could be many items, we want to arrange the serial numbers in 2
columns per page (with each page listing 30 items). Hence if there are 31
to 60 items packed, the packing slip document will have 2 pages. If there
are 61 to 90 items, there will be 3 pages.

**Comment by MarcinZiabek** [comment]
So, does it mean that the height of each element in the grid is well known? And adjusted in such a way that there are exactly 15 items in each column?

**Comment by MarcinZiabek** [comment, marked as answer]
This code is far from being perfect but it is shows how described layout can be accomplished:

```csharp
var elements = Enumerable
    .Range(1, 100)
    .Select(x => $"Item {x}")
    .ToList();

container
    .Background(Colors.White)
    .Padding(25)
    .Stack(pageStack =>
    {
        while (elements.Any())
        {
            var pageElements = elements.Take(30).ToList();
            elements = elements.Skip(30).ToList();
            
            pageStack.Item().Row(row =>
            {
                row.RelativeColumn().Stack(stack =>
                {
                    pageElements
                        .Take(15)
                        .ToList()
                        .ForEach(x => stack.Item().Border(1).Padding(10).Text(x));
                });
                
                row.RelativeColumn().Stack(stack =>
                {
                    pageElements
                        .Skip(15)
                        .ToList()
                        .ForEach(x => stack.Item().Border(1).Padding(10).Text(x));
                });
            });
            
            pageStack.Item().PageBreak();
        }
    });
```

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
